site/content/posts/2021-08-27-supporting-ad-oidc-workflows.md

```diff
@@ -0,0 +1,115 @@
+
+   # The password of your Active Directory bind account.
+   password: "YOUR_PASSWORD"
+ ```
+
+You can also customize the userSearch and groupSearch as shown in the examples in our reference documentation [here] ({{< ref “docs/howto/configure-supervisor-with-activedirectory.md" >}})
+
+Here is an example of what the ID token claims will look like:
+
+```yaml
+  aud:
+  - pinniped-cli
+  auth_time: 1630094468
+  exp: 1630094589
+  groups:
+  - Mammals@activedirectory.test.example.com
+  - Marine Mammals@activedirectory.test.example.com
+  iat: 1630094469
+  iss: https://pinniped-supervisor-clusterip.supervisor.svc.cluster.local/some/path
+  jti: 191709eb-b2fd-47e0-97f4-a06c48330c3a
+  nonce: c772c414388482163515103110cfcdfc
+  rat: 1630094468
+  sub: ldaps://activedirectory.test.example.com:636?base=DC%3Dactivedirectory%2CDC%3Dtest%2CDC%3Dexample%2CDC%3Dcom&sub=04030201-0605-0807-0910-111213141516
+  username: pinny@activedirectory.test.example.com
+ ```
+
+## OIDC CLI-based workflows
+
+In v0.10.0 we included support for Non-Interactive Password based LDAP logins to support CI/CD workflows. In this release, we extend the same capabilities to OIDC logins by using OIDC Password Grant. If the OIDC provider server supports the OAuth 2.0 resource owner password credentials grant, then you may optionally choose to configure `allowPasswordGrant` to `true` to allow clients to perform this type of authentication. Clients will be prompted for their username and password on the command-line without opening a browser window.
+It is important to note that [Resource Owner Password Credentials Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-4.3) from OAuth 2.0 is generally considered unsafe and should only be used when there is a trust relationship between the client and resource owner as it exposes client credentials to the resource owner. Refer to Security Best practices [here](https://datatracker.ietf.org/doc/html/rfc6749#section-4.3) However, it could be useful for use cases, such as for CI/CD where you may be authenticating to the Kubernetes cluster using an OIDC service account.
+
+### How this works with Pinniped
+A few considerations while configuring this on the cluster:
+
+Confirm that Multi-factor authentication is not intended to be used on the cluster
+Pinniped CLI running on your workstation and the Pinniped Supervisor backend are trusted to handle your password
+
+With the new functionality, Users initiate  pinniped get kubeconfig with a new argument --upstream-identity-provider-flow=”cli_password” to indicate their intent to use Password grant auth flow for logging into the upstream OIDC provider. By default, if no argument is specified this will follow the Browser-based auth flow. This way older Pinniped CLI versions will default to using Browser-based auth and the default for older Supervisor versions with newer CLI versions will also be Browser-based authentication.
```

Comment:
Looks like the backticks were lost around `pinniped get kubeconfig`?

Reply:
`”cli_password”` looks like it might have the wrong kinds of double quote characters around it too, similar to the copy/pate problem mentioned in my comment above.

Reply:
Resolved this in new commit